File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0024/24.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(0);
        dummy.next=  head;
        ListNode pre = dummy;
        while (pre.next != null && pre.next.next != null) {
            ListNode a = pre.next;
            ListNode b = pre.next.next;
            ListNode tmp = b.next;
            pre.next = b;
            b.next = a;
            a.next = tmp;
            pre = a;
        }
        return dummy.next;
    }
}

File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0024/24.cpp
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* pre = dummy;
        while (pre->next != nullptr && pre->next->next != nullptr) {
            ListNode* a = pre->next;
            ListNode* b = pre->next->next;
            ListNode* tmp = b->next;
            pre->next = b;
            b->next = a;
            a->next = tmp;
            pre = a;
        }
        return dummy->next;
    }
};